Upstream: Yes

From 542e8c9488890c87a39c3a128275ad8035061552 Mon Sep 17 00:00:00 2001
From: Jan Beich <jbeich@FreeBSD.org>
Date: Fri, 18 Sep 2020 15:34:46 +0200
Subject: [PATCH] types: avoid casting to non-standard __off64_t

off_t is always 64-bit on BSDs but only FreeBSD added off64_t compatibility.
Meson passes -D_FILE_OFFSET_BITS=64 by default, so there's no need for off64_t.
---
 src/types/buffer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/types/buffer.c b/src/types/buffer.c
index ba1c941..e6f80f3 100644
--- a/src/types/buffer.c
+++ b/src/types/buffer.c
@@ -78,7 +78,7 @@ static bool get_shm_fd (int *fd, size_t size)
 		if ( *fd >= 0 )
 		{
 			shm_unlink(name);
-			if ( ftruncate(*fd, (__off64_t)size) < 0 )
+			if ( ftruncate(*fd, (off_t)size) < 0 )
 			{
 				close(*fd);
 				return false;
-- 
2.26.2

